File: problemas/matematica/1240/1240.py
def verificar(A, B):
    tamanho = len(B)
    ultimos_digitos = A[-tamanho:]

    return True if ultimos_digitos == B else False

def main():
    n = int(input())
    for _ in range(n):
        a, b = map(str, input().split())
        print("encaixa" if verificar(a, b) else "nao encaixa")
main()

File: problemas/matematica/1240/1240.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";

process.stdin.on('data', (chunk) => {
    input += chunk;
});

process.stdin.on('end', () => {
    function verificar (A, B) {
        const tamanho = B.length;
        const ultimosDigitos = A.slice(-tamanho);

        return ultimosDigitos === B;
    }

    let lines = input.split('\n');
    let N = parseInt(lines.shift().trim());

    for (let i = 0; i < N; ++i) {
        let [A, B] = lines.shift().trim().split(' ').map(String);

        console.log(verificar(A, B)? "encaixa": "nao encaixa");
    }
});
